src/CollectorGuess.cpp

```diff
@@ -50,6 +50,10 @@ bool isNumber(const std::string& x, LocaleInfo* pLocale) {
   if (x[0] == '0' && x.size() > 1 && x[1] != pLocale->decimalMark_)
     return false;
 
+  // Only a - or decimal mark
+  if (x.size() == 1 && x[0] == '-' || x[0] == pLocale->decimalMark_)
+    return false;
+
```

Comment:
I think the problem with only fixing it here is that then `parse_numeric(c(".". "-"))` might give incorrect values.